lwt/tls_lwt.ml

```diff
@@ -31,10 +31,43 @@ module Lwt_cs = struct
     | cs -> write fd cs >>= o (write_full fd) (Cstruct.shift cs)
 end
 
+module Lwt_fd = struct
+  type t = {
+    read  : Cstruct.t -> int Lwt.t ;
+    write : Cstruct.t -> unit Lwt.t ;
+    close : unit -> unit Lwt.t ;
+  }
```

Comment:
I didn't want to mess with the carefully crafted `write`/`close` implementation present for file descriptors, but perhaps using a pair of `input/output Lwt_io.channel`s would work instead of a new abstraction? (I'm not sure of the impact, because the doc for lwt channels mentions that they do buffering)